contributor_docs/webgl_mode_architecture.md

```diff
@@ -242,22 +242,49 @@ It also has the following per-vertex attributes:
+```mermaid
+---
+title: p5.js WebGL Classes
+---
+classDiagram
+    class Base["p5.Renderer"] {
+    }
+    class P2D["p5.Renderer2D"] {
+    }
+    class WebGL["p5.RendererGL"] {
+    }
+    class Geometry["p5.Geometry"] {
+    }
+    class Shader["p5.Shader"] {
+    }
+    class Texture["p5.Texture"] {
+    }
+    class Framebuffer["p5.Framebuffer"] {
+    }
+    Base <|-- P2D
+    Base <|-- WebGL
+    WebGL "*" o-- "*" Geometry
+    WebGL "1" *-- "*" Shader
+    WebGL "1" *-- "*" Texture
+    WebGL "1" *-- "*" Framebuffer
+```
+
+The entry point to most WebGL code is through **p5.RendererGL**. Top-level p5.js functions are passed to the current renderer. Both 2D and WebGL modes have renderer classes that conform to a common `p5.Renderer` interface. Immediate mode and retained mode functions are split up into **p5.RendererGL.Immediate.js** and **p5.RendererGL.Retained.js**.
+
+Within the renderer, references to models are stored in the **retainedMode.geometry** map. Each value is an object storing the buffers of a **p5.Geometry**. When calling `model(yourGeometry)` for the first time, the renderer adds an entry in the map. It then stores references to the geometry's GPU resources there. If you draw a `p5.Geometry` to the main canvas and also to a WebGL `p5.Graphics`, it will have entries in two renderers.
+
+Each material is represented by a **p5.Shader.** You set the current shader in the renderer via the `shader(yourShader)` function. This class handles compiling shader source code and setting shader uniforms.
+
+When setting a shader uniform, if the uniform type is an image, then the renderer creates a `p5.Texture` for it. Each `p5.Image`, `p5.Graphics`, `p5.MediaElement`, or `p5.Framebuffer` asset will get one. It is what keeps track of the image data's representation on the GPU. Before using the asset in a shader, p5.js will send new data to the GPU if necessary. For images, this happens when a user has manually updated the pixels of an image. This happens every frame for assets with data that may have changed each frame, such as a video or a `p5.Graphics`.
+
+Textures corresponding to **p5.Framebuffer** objects are unique. Framebuffers are like graphics: they represent surfaces that can be drawn to. Unlike `p5.Graphics`, `p5.Framebuffer`s live entirely on the GPU. If one uses a `p5.Graphics` as a texture in a shader, the data needs to be transferred to and from the CPU. This can often be a performance bottleneck. In contrast, when drawing to a `p5.Framebuffer`, you draw directly to its GPU texture. Because of this, no extra data transfer is necessary. WebGL mode tries to use `p5.Framebuffer`s over `p5.Graphics` where possible for this reason.
```

Comment:
Syntax highlighting: **p5.Framebuffer** should probably be `p5.Framebuffer`

Consistency: `p5.Framebuffer`s are like graphics: ...